qa/workunits/rbd/rbd-nbd.sh

```diff
@@ -442,4 +445,25 @@ _sudo rbd device detach ${DEV} --device-type nbd
 expect_false get_pid ${POOL}
 DEV=
 
+# test discard granularity with journaling
+rbd feature enable ${POOL}/${IMAGE} journaling
+bs=4096
+rbd config global set client rbd_discard_granularity_bytes ${bs}
+DEV=`_sudo rbd device --device-type nbd map ${POOL}/${IMAGE}`
+get_pid ${POOL}
+_sudo blkdiscard --offset 0 --length $((bs+512)) ${DEV}
```

Comment:
I did not do exactly this, but rather 8192-1024, 8192-512, 8192 since 8192 would be the one hitting the alignment. In sorted order from smallest to largest.

I changed rbd config to actually set the parameter on image instead of global since it was only a fluke that it worked. I validated it and rbd config global set global did not affect the discard_granularity_bytes setting. Both rbd config pool and rbd config image did however affect it live.

Sadly I can't set rbd_discard_granularity_bytes 0 live, since it's not allowed, and rbd_skip_partial_discard does not change rbd_discard_granularity_bytes live either. Had it been possible to set rbd_discard_granularity_bytes to 0 directly (still to the power of 2 I guess?) that would mean we could have fixed this issue by issuing `rbd config pool set rbd rbd_discard_granularity_bytes 0`, in one sweep.